src/kani-verifier/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+const TARGET: &str = env!("TARGET");
+
+/// Typically `~/.kani/kani-1.x/`
+fn kani_dir() -> PathBuf {
+    home::home_dir()
+        .expect("Couldn't find home dir?")
+        .join(".kani")
+        .join(format!("kani-{}", VERSION))
+}
+
+/// The filename of the release bundle
+fn download_filename() -> String {
+    format!("kani-{}-{}.tar.gz", VERSION, TARGET)
+}
+
+/// Helper to find the download URL for this version of Kani
+fn download_url() -> String {
+    let tag: &str = &format!("kani-{}", VERSION);
+    let file: &str = &download_filename();
+    format!("https://github.com/model-checking/kani/releases/download/{}/{}", tag, file)
+}
+
+/// Effectively the entry point (i.e. `main` function) for both our proxy binaries.
+pub fn proxy(bin: &str) -> Result<()> {
+    // In an effort to keep our dependencies minimal, we do the bare minimum argument parsing
+    let args: Vec<_> = std::env::args_os().collect();
+    if args.len() >= 2 && args[1] == "setup" {
+        if args.len() >= 4 && args[2] == "--use-local-bundle" {
+            setup(Some(args[3].clone()))
+        } else {
+            setup(None)
+        }
+    } else {
+        if !appears_setup() {
+            setup(None)?;
+        }
+        exec(bin)
+    }
+}
+
+/// Fast check to see if we look setup already
+fn appears_setup() -> bool {
+    kani_dir().exists()
+}
+
+/// Sets up Kani by unpacking/installing to `~/.kani/kani-VERSION`
+fn setup(use_local_bundle: Option<OsString>) -> Result<()> {
+    let kani_dir = kani_dir();
+    // e.g. `~/.kani/`
+    let base_dir = kani_dir.parent().expect("No base directory?");
+
+    println!("[0/6] Running Kani first-time setup...");
+
+    println!("[1/6] Ensuring the existence of: {}", base_dir.display());
+    std::fs::create_dir_all(&base_dir)?;
+
+    if let Some(pathstr) = use_local_bundle {
+        let path = Path::new(&pathstr).canonicalize()?;
+        println!("[2/6] Installing local Kani bundle: {}", path.display());
+        Command::new("tar").arg("zxf").arg(&path).current_dir(base_dir).run()?;
+
+        // when given a local bundle, it's often "-latest" but we expect "-1.0" or something. Hack it up.
+        let file = path.file_name().expect("has filename").to_string_lossy();
+        let components: Vec<_> = file.split('-').collect();
+        let expected_dir = format!("{}-{}", components[0], components[1]);
+
+        std::fs::rename(base_dir.join(expected_dir), &kani_dir)?;
+    } else {
+        let filename = download_filename();
+        println!("[2/6] Downloading Kani release bundle: {}", &filename);
+        let bundle = base_dir.join(filename);
+        Command::new("curl").args(&["-sSLf", "-o"]).arg(&bundle).arg(download_url()).run()?;
+
+        Command::new("tar").arg("zxf").arg(&bundle).current_dir(base_dir).run()?;
+
+        std::fs::remove_file(bundle)?;
+    }
+
+    let toolchain_version = std::fs::read_to_string(kani_dir.join("rust-toolchain"))
```

Comment:
I would prefer if we just upload the rust toolchain file here. If you are trying to avoid any sort of parsing, I would recommend using a different name. The `rust-toolchain` file is a legacy name used by `rustup` for configuration files that have specific file format specifications.

https://rust-lang.github.io/rustup/overrides.html?highlight=rust-toolchain.toml#the-toolchain-file